### PR TITLE
CPV,PHS: modernize cell, digit and cluster readers

### PR DESCRIPTION
**Pull request by sevdokim**
Hello! In this PR new digit & cluster reader for cpv and new digit & cell readers for phs are added. This readers needed in MC production for connecting QC. Old readers cannot be used because they produce one more empty TF when finish reading, and this empty TF crashes QC.

### CODE REVIEW

**Comment by sevdokim** [comment]
Hi @shahor02 
I got fails of trivial checks with error
`
fatal: couldn't find remote ref pull/12060/head
`
What can it be?

**Comment by shahor02** [comment]
pinging to @TimoWilken

**Comment by sevdokim** [comment]
Force-pushing triggered checks again. This time they were successful.

File: Detectors/PHOS/workflow/src/CellConverterSpec.cxx [hunk 1]
```diff
@@ -37,10 +37,12 @@ void CellConverterSpec::init(framework::InitContext& ctx)
...
-  LOG(debug) << "[PHOSCellConverter - run] called";
-  auto dataref = ctx.inputs().get("digits");
-  auto const* phosheader = o2::framework::DataRefUtils::getHeader<o2::phos::PHOSBlockHeader*>(dataref);
-  if (!phosheader->mHasPayload) {
+  // LOG(debug) << "[PHOSCellConverter - run] called";
+  // auto dataref = ctx.inputs().get("digits");
+  // auto const* phosheader = o2::framework::DataRefUtils::getHeader<o2::phos::PHOSBlockHeader*>(dataref);
+  // if (!phosheader->mHasPayload) {
+  auto digits = ctx.inputs().get<std::vector<o2::phos::Digit>>("digits");
+  if (!digits.size()) { // nothing to process
```
Comment (peressounko) [inline Detectors/PHOS/workflow/src/CellConverterSpec.cxx]: @sevdokim , what if we have event without digits? Probably it is better to use TriggerRecords - they will be non-zero in such events

File: Detectors/PHOS/workflow/src/ClusterizerSpec.cxx [hunk 2]
```diff
@@ -72,9 +72,11 @@ void ClusterizerSpec::run(framework::ProcessingContext& ctx)
...
     LOG(debug) << "PHOSClusterizer - run on digits called";
 
-    auto dataref = ctx.inputs().get("digits");
-    auto const* phosheader = o2::framework::DataRefUtils::getHeader<o2::phos::PHOSBlockHeader*>(dataref);
-    if (!phosheader->mHasPayload) {
+    // auto dataref = ctx.inputs().get("digits");
+    // auto const* phosheader = o2::framework::DataRefUtils::getHeader<o2::phos::PHOSBlockHeader*>(dataref);
+    // if (!phosheader->mHasPayload) {
+    auto digits = ctx.inputs().get<std::vector<o2::phos::Digit>>("digits");
+    if (!digits.size()) { // nothing to process
```
Comment (peressounko) [inline Detectors/PHOS/workflow/src/ClusterizerSpec.cxx]: @sevdokim , similar comment: should we created empty list of clusters for events with empty list of digits?

File: Detectors/PHOS/workflow/src/CellConverterSpec.cxx (hunk 1 above)
Comment (sevdokim) [reply Detectors/PHOS/workflow/src/CellConverterSpec.cxx]: @peressounko can such case even be?

Comment (sevdokim) [reply Detectors/PHOS/workflow/src/CellConverterSpec.cxx]: but yes, I agree if we have digit but no corresponding trigger record that can be a problem

**Review by peressounko — DISMISSED** [review]

**Review by shahor02 — COMMENTED** [review]
Thanks, could you also use new readers here: https://github.com/AliceO2Group/AliceO2/blob/500326b37dc5e3f22b16092b84f41c27ba125b9b/Detectors/GlobalTrackingWorkflow/helpers/src/InputHelper.cxx#L166-L172

**Comment by sevdokim** [comment]
hi @shahor02 
yes, of cause. Can you please tell me how to check that GlobalTrackingWorkflow works as expected after changing readers?

**Comment by shahor02** [comment]
Global tracking workflow is just a collection of workflows. The InputHelper allows to start set of readers on demand. So, if your previous reader workflows were doing an extra run cycle, just substitute relevant old `getXXXreaderSpec` by that of the new readers.

**Comment by sevdokim** [comment]
Hi @shahor02 
thanks. I putted new readers to InputHelper.cxx
Checks are passed except masOS build but I think it is not relevant.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**